clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -288,4 +300,26 @@ private Duration getTimeoutDuration(EhcacheOperationMessage message) {
       }
     }
   }
+
+  private static void waitForLatch(long nanos, CountDownLatch latch) {
+    boolean interrupted = false;
+    long deadlineTimeout = System.nanoTime() + nanos;
+    try {
+      while (true) {
+        try {
+          long timeRemaining = deadlineTimeout - System.nanoTime();
+          if (timeRemaining <= 0) {
+            return;
+          }
+          latch.await(timeRemaining, TimeUnit.NANOSECONDS);
```

Comment:
I don't really like the latch hanging around even though it will rarely be useful... Maybe we should just sleep and loop here until `initComplete` or timeout. To remove the latch.

Reply:
How about wait/notify? A sleep/wakeup loop may not be responsive enough.

Also since this is an cluster tier entity, we should not be expecting more than a few 100s max, right?